src/csg_tree.cpp

```diff
@@ -109,9 +109,51 @@ std::shared_ptr<CsgNode> CsgLeafNode::Transform(const mat3x4 &m) const {
 CsgNodeType CsgLeafNode::GetNodeType() const { return CsgNodeType::Leaf; }
 
 std::shared_ptr<CsgLeafNode> ImplToLeaf(Manifold::Impl &&impl) {
+#ifdef MANIFOLD_DEBUG
+  if (impl.IsSelfIntersecting()) {
```

Comment:
Maybe also put this behind the `intermediate_checks` flag? I don't think there's a safe way to test this in general.